networks/edsl_examples/reorgyolo.py

```diff
@@ -0,0 +1,189 @@
+#   -------------------------- Testing method-------------------------------------
+#   The reorg function provided here:
+#   https://gist.github.com/leimao/ece7217b5d07fe4e685c47af5e76744a
+#   is currently being used for local testing
+# TODO: write python unittests
+# TODO: write backend_test style test against pytorch implementation
+# TODO: remove test functions and python test code
+# TODO: write a note on handling the modulus operator in eDSL
+# TODO: write a note on how floor division works in eDSL
+
+import numpy as np
+
+import plaidml2 as plaidml
+import plaidml2.edsl as edsl
+import plaidml2.exec as plaidml_exec
+import plaidml2.op as plaidml_op
+
+
+def reorgyolo_comparison(arrayIn, batch, C, H, W, stride, forward=False):
+    arrayLen = len(arrayIn)
+    arrayOut = np.zeros(arrayLen)
+    print("C is " + str(C) + "stride is " + str(stride))
+    out_c = C // (stride * stride)
+    for b in range(batch):
+        for k in range(C):
+            for j in range(H):
+                for i in range(W):
+                    in_index = i + W * (j + H * (k + C * b))
+                    c2 = k % out_c
+                    offset = k // out_c
+                    w2 = i * stride + offset % stride
+                    h2 = j * stride + offset // stride
+                    out_index = int(w2 + W * stride * (h2 + H * stride * (c2 + out_c * b)))
+                    if forward:
+                        arrayOut[out_index] = arrayIn[in_index]
+                    else:
+                        arrayOut[in_index] = arrayIn[out_index]
+    return arrayOut
+
+
+def reorgyolo_comparison_nodivmod(arrayIn, batch, C, H, W, stride, forward=False):
+    arrayLen = len(arrayIn)
+    arrayOut = np.zeros(arrayLen)
+    print("C is " + str(C) + "stride is " + str(stride))
+    out_c = C // (stride * stride)
+    _c1_quotient_range = int(C // (out_c))
+    for n1 in range(batch):
+        for w1 in range(W):
+            for h1 in range(H):
+                for c2 in range(out_c):
+                    for _w2_quotient in range(_c1_quotient_range // stride):
+                        for _w2 in range(stride):
+                            _c1 = _w2 + _w2_quotient * stride
+                            c1 = c2 + (_c1 * out_c)
+                            in_index = w1 + W * (h1 + H * (c1 + C * n1))
+                            w2 = w1 * stride + _w2
+                            h2 = h1 * stride + _w2_quotient
+                            out_index = int(w2 + W * stride * (h2 + H * stride *
+                                                               (c2 + out_c * n1)))
+                            if forward:
+                                arrayOut[out_index] = arrayIn[in_index]
+                            else:
+                                arrayOut[in_index] = arrayIn[out_index]
+    return arrayOut
+
+
+def reorgyolo(I, stride, forward):
+    dims = I.shape.int_dims
+    N = dims[0]
+    C = dims[1]
+    H = dims[2]
+    W = dims[3]
```

Comment:
These aren't needed, `I.bind_dims()` does the same thing. BTW, this won't work in plaidml-v1, `shape` has been dropped from `Tensor`.